opentelekomcloud/services/iam/data_source_opentelekomcloud_identity_project_v3.go

```diff
@@ -106,10 +114,18 @@ func dataSourceIdentityProjectV3Read(_ context.Context, d *schema.ResourceData,
 			"Please change your search criteria and try again.")
 	}
 
+	var NewFilteredProjects []projects.Project
 	if len(filteredProjects) > 1 {
-		log.Printf("[DEBUG] Multiple results found: %#v", allProjects)
-		return fmterr.Errorf("your query returned more than one result")
+		projectID := config.HwClient.ProjectID
+		for _, p := range filteredProjects {
+			if p.ID != projectID {
+				continue
+			}
+			NewFilteredProjects = append(NewFilteredProjects, p)
```

Comment:
There can be only a single project with such ID, you can exit the loop when it's found.
In this case you can do like this:
```go
filteredProjects = []projects.Project{*p}
```